[Noetic] CMake error: keyword signature for target_link_libraries has already been used

From job: https://build.ros.org/view/Nbin_uF64/job/Nbin_uF64__sot-core__ubuntu_focal_amd64__binary/86/console

sot-core, sot-dynamic-pinocchio, and sot-tools aren't building on the ROS Noetic buildfarm. These packages will be missing from the next sync unless they're fixed. If you do fix them, please @sloretz on the ros/rosdistro PR so I know to merge it during the sync freeze.

```
23:13:45 CMake Error at src/CMakeLists.txt:121 (target_link_libraries):
23:13:45   The keyword signature for target_link_libraries has already been used with
23:13:45   the target "sot-core-sot-wrap".  All uses of target_link_libraries with a
23:13:45   target must be either all-keyword or all-plain.
23:13:45 
23:13:45   The uses of the keyword signature are here:
23:13:45 
23:13:45    * /opt/ros/noetic/share/eigenpy/cmake/python.cmake:440 (target_link_libraries)
23:13:45    * /opt/ros/noetic/share/eigenpy/cmake/python.cmake:442 (target_link_libraries)
23:13:45    * cmake/boost.cmake:251 (TARGET_LINK_LIBRARIES)
23:13:45    * /opt/ros/noetic/share/eigenpy/cmake/python.cmake:448 (target_link_libraries)
```


Possibly related to https://github.com/stack-of-tasks/eigenpy/pull/307 ?
It seems to be related to this change upstream: https://github.com/jrl-umi3218/jrl-cmakemodules/commit/432dfe73cbcf4a9bba238f650f0ef191c79496a3

Unfortunately even with the new version of eigenpy the sot-core job is still failing with the same error.

https://build.ros.org/job/Nbin_uF64__sot-core__ubuntu_focal_amd64__binary/87/consoleText

```
Setting up ros-noetic-eigenpy (2.7.12-1focal.20220824.190355) ...
```

```
CMake Error at src/CMakeLists.txt:121 (target_link_libraries):
  The keyword signature for target_link_libraries has already been used with
  the target "sot-core-sot-wrap".  All uses of target_link_libraries with a
  target must be either all-keyword or all-plain.

  The uses of the keyword signature are here:

   * /opt/ros/noetic/share/eigenpy/cmake/python.cmake:440 (target_link_libraries)
   * /opt/ros/noetic/share/eigenpy/cmake/python.cmake:442 (target_link_libraries)
   * /opt/ros/noetic/share/eigenpy/cmake/boost.cmake:293 (target_link_libraries)
   * /opt/ros/noetic/share/eigenpy/cmake/python.cmake:448 (target_link_libraries)
```

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -118,7 +118,7 @@ FOREACH(plugin ${plugins})
         ${LIBRARY_NAME} ${PROJECT_NAME}-${PYTHON_LIBRARY_NAME}-wrap
         MODULE_HEADER "${CMAKE_CURRENT_SOURCE_DIR}/${plugin}-python.h")
     endif()
-    target_link_libraries(${PROJECT_NAME}-${PYTHON_LIBRARY_NAME}-wrap eigenpy::eigenpy)
+    target_link_libraries(${PROJECT_NAME}-${PYTHON_LIBRARY_NAME}-wrap PUBLIC eigenpy::eigenpy)
   ENDIF(BUILD_PYTHON_INTERFACE)
 ENDFOREACH(plugin)
 

